### DISCUSSION TITLE
Onestop ID link

### DISCUSSION THREAD

**Discussion by piperswe** [transitland/transitland#367] [Q&A] [answered]
Is there an endpoint that takes an arbitrary Onestop ID in its URL and redirects to the entity's page, deciding the correct route using the first character? For example, a route `/onestop/$1` such that `/onestop/o-9-amtrak` redirects to `/operators/o-9-amtrak`, `/onestop/r-gcpv-piccadilly` redirects to `/routes/r-gcpv-piccadilly`, `/onestop/s-u1h2f9gxy5-aachenhbf` redirects to `/stops/s-u1h2f9gxy5-aachenhbf`, `/onestop/f-9-amtrak~amtrakcalifornia~amtrakcharteredvehicle` redirects to `/feeds/f-9-amtrak~amtrakcalifornia~amtrakcharteredvehicle`, etc. I'm looking to add a Wikidata property for an entity's Onestop ID, and having one URL that takes a Onestop ID and redirects to the appropriate page makes URL generation possible.

**Comment by drewda** [comment, marked as answer]
The v1 Datastore API does have an endpoint to take any Onestop ID. Try for example:

    https://transit.land/api/v1/onestop_id/o-9q9-bart?apikey=xxx

The Datastore API is deprecated, so we don't recommend using this going forward. But we would be glad to consider adding a similar endpoint to the v2 REST API.

**Reply by piperswe** [reply]
I'm more looking for something on the website side, not the API - I'm looking to be able to link people to look at the entity on transit.land

**Reply by drewda** [reply]
O.K., that makes sense.

We've added some functionality so now you can use URLs like: https://www.transit.land/onestop-id/f-9q9-bart

More info at https://www.transit.land/documentation/onestop-id-scheme/#look-up-any-entity-by-onestop-id
